robosherlock/src/filter/src/PointCloudFilter.cpp

```diff
@@ -81,14 +87,38 @@ class PointCloudFilter : public DrawingAnnotator
     MEASURE_TIME;
     outInfo("process start");
     rs::SceneCas cas(tcas);
+    rs::Scene scene = cas.getScene();
     pcl::PointCloud<PointT>::Ptr cloud_ptr(new pcl::PointCloud<PointT>);
-    (new pcl::PointCloud<PointT>);
 
     cas.get(VIEW_CLOUD, *cloud_ptr);
     cas.get(VIEW_COLOR_IMAGE, rgb_);
 
+    pcl::PointCloud<PointT>::Ptr cloud_transformed(new pcl::PointCloud<PointT>);
+    tf::Transform viewport;
+
+    if(transform_cloud) {
+      //@Todo This has nothing to do with base frame yet...
+      outInfo("Transforming Point Cloud to Viewport space");
+      tf::StampedTransform camToWorld;
+      tf::StampedTransform worldToCam;
+      camToWorld.setIdentity();
+      if(scene.viewPoint.has()) {
```

Comment:
you could replace this with a simple tf lookupTransform; [see here example](https://wiki.ros.org/tf/Tutorials/Writing%20a%20tf%20listener%20%28C%2B%2B%29) and use  a param for setting the frame;